website/docs/guides/serverless-with-aws-lambda-and-api-gateway.html.md

```diff
@@ -0,0 +1,579 @@
+        },
+        body: "<p>Hello world!</p>",
+    };
+    callback(null, response);
+};
+```
+
+The above is the simplest possible Lambda function for use with API Gateway,
+returning a hard-coded "Hello world!" response in the object structure that
+API Gateway expects.
+
+From your command prompt, change to the directory containing that file and
+add it to a zip file in the parent directory:
+
+```
+$ cd example
+$ zip ../example.zip main.js
+  adding: main.js (deflated 33%)
+$ cd ..
+```
+
+In a real build and deploy scenario we would have an S3 bucket set aside for
+staging our archive and would use this to "hand off" these artifacts between
+the build and deploy process. For the sake of this tutorial we will create
+a temporary S3 bucket using the AWS CLI. S3 bucket names are globally unique,
+so you may need to change the `--bucket=` argument in the following example
+and substitute your new bucket name throughout the rest of this tutorial.
+
+```
+$ aws s3api create-bucket --bucket=terraform-serverless-example --region=us-east-1
+```
+
+You can now upload your build artifact into this S3 bucket:
+
+```
+$ aws s3 cp example.zip s3://terraform-serverless-example/v1.0.0/example.zip
+```
+
+A version number is included in the object path to identify this build. Later
+we will demonstrate deploying a new version, which will create another
+separate object.
+
+## Creating the Lambda Function
+
+With the source code artifact built and uploaded to S3, we can now write our
+Terraform configuration to deploy it. In a new directory, create a file
+named `lambda.tf` containing the following configuration:
+
+```hcl
+provider "aws" {
+  region = "us-east-1"
+}
+
+resource "aws_lambda_function" "example" {
+  function_name = "ServerlessExample"
+
+  # The bucket name as created earlier with "aws s3api create-bucket"
+  s3_bucket = "terraform-serverless-example"
+  s3_key    = "v1.0.0/example.zip"
+
+  # "main" is the filename within the zip file (main.js) and "handler"
+  # is the name of the property under which the handler function was
+  # exported in that file.
+  handler = "main.handler"
+  runtime = "nodejs6.10"
+
+  role = "${aws_iam_role.lambda_exec.arn}"
+}
+
+# IAM role which dictates what other AWS services the Lambda function
+# may access.
+resource "aws_iam_role" "lambda_exec" {
+  name = "serverless_example_lambda"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+```
+
+Each Lambda function must have an associated IAM role which dictates what
+access it has to other AWS services. The above configuration specifies a role
+with no access policy, effectively giving the function no access to any
+AWS services, since our example application requires no such access.
+
+Apply the above configuration as usual:
```

Comment:
Shouldn't we also mention `terraform init` here?

Reply:
A fair point... I assumed the user already knows how to do that, but there's not much harm in pointing it out again here to make sure.